cpp/src/arrow/compute/kernels/base_arithmetic_internal.h

```diff
@@ -60,6 +60,11 @@ struct Add {
   static enable_if_decimal_value<T> Call(KernelContext*, Arg0 left, Arg1 right, Status*) {
     return left + right;
   }
+
+  template <typename T>
+  static constexpr T Identity() {
```

Comment:
The term identity is from the mathematical notation monoid. For any associative binary operation, identity is definded as Op(identity, x) = x for all x.